nx-X11/programs/Xserver/hw/nxagent/Init.c

```diff
@@ -192,7 +192,7 @@ int nxagentX2go;
  * Checking if agent is x2go agent
  */
 
-void checkX2goAgent()
+void checkX2goAgent(void)
 {
   extern const char *nxagentProgName;
```

Comment:
I have noticed that you can pass product=something in the options file. So I suggest x2go should add product=x2go and _that_ shoudl be checked here instead of the progname.